src/app/time-table/time-table.component.ts

```diff
@@ -1,17 +1,25 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { OperationService } from '../shared/operation.service';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'app-time-table',
   templateUrl: './time-table.component.html',
   styleUrls: ['./time-table.component.css']
 })
-export class TimeTableComponent implements OnInit {
-  result = this.operationService.counterValue;
+export class TimeTableComponent implements OnInit, OnDestroy {
+  result!: number;
```

Comment:
be careful when using non-null assertions. I would always recommend a default value instead of non-null

Reply:
yes it was better to go with initializations instead of non null.